### ISSUE TITLE
Plugins not executing in IRC

### ISSUE THREAD

**Issue by Macr0Nerd** [Abdur-rahmaanJ/honeybot#56]
**Describe the bug**
Testing the bot on my personal IRC channel, I discovered that the bot would not detect plugins and typing things such as `.test` and `.greet` would return nothing despite appearing in the python terminal. Was testing for Issue #36.

**To Reproduce**
1. Download and set up bot (username: `hb_mn00`)
2. Change autoconnect channels
3. Run `main.py`
4. Test commands such as `.test`, `.greet`, and `.btc`

**Expected behavior**
The bot to return the plugins output, such as `hoo` or the current price of bitcoin.

**Screenshots**
![image](https://user-images.githubusercontent.com/16966836/54574505-8d9cfb00-49be-11e9-8aab-523a027bfcf8.png)
![image](https://user-images.githubusercontent.com/16966836/54574526-a7d6d900-49be-11e9-9373-5dafaed7edd1.png)
*Working on adding more screenshots*

**Desktop (please complete the following information):**
 - OS: Fedora 29 Linux x86_64
 - IDE: Pycharm Community 2019.3
 - IRC: Hexchat, Freenode

**Potential Solutions**
 - ~~I have been looking into a similar import structure as with `pluginInfo.py` using dynamic improt structure.~~

**Comment by Abdur-rahmaanJ** [comment]
testing in some 45 mins ...

**Comment by Macr0Nerd** [comment]
Currently debugging @Abdur-rahmaanJ. My idea for importing didn't seem to work and I'm currently going through line by line. May not be able to continue until tomorrow, though.

**Comment by Abdur-rahmaanJ** [comment]
some work, not all, will continue tests

**Comment by Macr0Nerd** [comment]
![image](https://user-images.githubusercontent.com/16966836/54651609-a4585600-4a81-11e9-98c1-f5e3988f9aec.png)
@Abdur-rahmaanJ I'm not getting anything at all. What external dependencies might I need for `main.py`? It could be that

**Comment by Macr0Nerd** [comment]
@Abdur-rahmaanJ so after some long debugging, I've discovered my main problem is that I wasn't setting plugins with to_load. I'm now encountering the error that it is missing `info`. 

![image](https://user-images.githubusercontent.com/16966836/54654002-b8548580-4a8a-11e9-8252-e90bc5ea41ff.png)

    def load_plugins(self, list_to_add):
        print("\033[0;36mLoading plugins...\033[0;0m")
        try:
            to_load = []
            with open('PLUGINS.conf', 'r') as f:
                to_load = f.read().split('\n')
                to_load = list(filter(lambda x: x != '', to_load))
            for file in to_load:
                module = importlib.import_module('plugins.'+file)
                obj = module.Plugin
                list_to_add.append(obj)

            self.plugins = list_to_add
            print("\033[0;32mLoaded plugins...\033[0;0m")
        except ModuleNotFoundError as e:
            print('module not found', e)

**Comment by Macr0Nerd** [comment]
@Abdur-rahmaanJ so far it's almost worked. It is just missing the `info` argument. Otherwise, I fixed

**Comment by Macr0Nerd** [comment]
Accidentally clicked the wrong button sorry.

**Comment by Macr0Nerd** [comment]
@Abdur-rahmaanJ So I'm getting this error with certain plugins. It is telling me

     error quote plugin 'Bot_core' object has no attribute 'quote'

It does this with a lot of the plugins. I haven't tested them all, but so far only greet and bitcoin work.

**Comment by Macr0Nerd** [comment]
@Abdur-rahmaanJ so I found what seems like a fix to the issue I described in my previous comment. When we use `self.func()` in a Plugin class, it treats `self` as `Bot_core`, not `Plugin`. I fixed this for some of the plugins with `Plugins.func(self)`. It is far from clean, but it works as far as I can tell, and isn't code breaking. I'm going to do some further testing and fix some stuff like the weather plugin.
